### PR TITLE
Add a --dry-run option to `argo install`. Remove CRD creation from controller startup

### PR DESCRIPTION
**Pull request by jessesuen**
This resolves issue #658.

This change adds a new flag `--dry-run` to `argo install` which prints the kube manifests instead of creating them in kubernetes. These manifests can be passed to kubectl so the following is now possible:
```
$ argo install --dry-run | kubectl create -f -
```

It also removes CRD creation from the workflow controller, which is the wrong place to do CRD creation since it increases the required RBAC privileges of the controller more than what is necessary.

### CODE REVIEW

File: cmd/workflow-controller/main.go [hunk 1]
```diff
@@ -23,6 +21,7 @@ import (
 )
 
 const (
+	// CLIName is the name of the CLI
```
Comment (shrinandj) [inline cmd/workflow-controller/main.go]: Any reason the cli is called "workflow-controller"?

**Comment by shrinandj** [comment]
Can you add an e2e test that will try to install Argo with the `--dry-run` option? Basically, the test will ensure that the K8s cluster is unchanged because of the `--dry-run` option.

File: cmd/workflow-controller/main.go (hunk 1 above)
Comment (jessesuen) [reply cmd/workflow-controller/main.go]: Just wanted to match the container image name and deployment name. I believe ingress-controller is the same way.

**Comment by jessesuen** [comment]
e2e test added

**Pull request merged**